app/src/test/java/cm/aptoide/pt/app/AppViewManagerTest.java

```diff
@@ -0,0 +1,717 @@
+
+    appViewManager =
+        new AppViewManager(installManager, downloadFactory, appCenter, reviewsManager, adsManager,
+            storeManager, flagManager, storeUtilsProxy, aptoideAccountManager, appViewConfiguration,
+            preferencesManager, downloadStateParser, appViewAnalytics, notificationAnalytics,
+            installAnalytics, limit, socialRepository);
+
+    //When the presenter ask for an App and the AppView was initialized with arguments other than appId, md5 or uniqueName
+    //And a result is returned with success
+    when(appCenter.loadDetailedApp("", "")).thenReturn(Single.just(detailedAppRequestResult));
+    when(store.getId()).thenReturn((long) 1);
+    when(storeManager.isSubscribed(anyLong())).thenReturn(Observable.just(true));
+
+    AppViewViewModel appViewViewModel = appViewManager.loadAppViewViewModel()
+        .toBlocking()
+        .value();
+
+    //Then the correct loadDetailedApp should be called
+    verify(appCenter).loadDetailedApp("", "");
+    //And a AppViewViewModel should be returned with a not null app, with no loading and no errors
+    Assert.assertNotNull(appViewViewModel.getAppId());
+    Assert.assertEquals(false, appViewViewModel.isLoading());
+    Assert.assertEquals(false, appViewViewModel.hasError());
+
+    //Test if app is cached
+    when(store.getName()).thenReturn("");
+    appViewViewModel = appViewManager.loadAppViewViewModel()
+        .toBlocking()
+        .value();
+    Assert.assertNotNull(appViewViewModel.getAppId());
+    Assert.assertEquals(false, appViewViewModel.isLoading());
+    Assert.assertEquals(false, appViewViewModel.hasError());
+
+    //Since there's a cached app there should not be any interactions with the appCenter
+    verifyZeroInteractions(appCenter);
+  }
+
+  @Test public void loadAppViewViewModelWithLoadingStateTest() {
+    DetailedAppRequestResult detailedAppRequestResult = new DetailedAppRequestResult(true);
+    AppViewConfiguration appViewConfiguration =
+        new AppViewConfiguration((long) 1, "anyString", "anyString", "", null, null, "", "", 0.0,
+            "", "");
+
+    appViewManager =
+        new AppViewManager(installManager, downloadFactory, appCenter, reviewsManager, adsManager,
+            storeManager, flagManager, storeUtilsProxy, aptoideAccountManager, appViewConfiguration,
+            preferencesManager, downloadStateParser, appViewAnalytics, notificationAnalytics,
+            installAnalytics, limit, socialRepository);
+
+    //When the presenter ask for an App
+    //And a result is returned
+    when(appCenter.loadDetailedApp((long) 1, "anyString", "anyString")).thenReturn(
+        Single.just(detailedAppRequestResult));
+
+    //Then an AppViewViewModel should be returned with a null app, with loading and no errors
+    appViewManager.loadAppViewViewModel()
+        .map(AppViewViewModel::isLoading)
+        .test()
+        .assertValue(true);
```

Comment:
Personally i would add another verification, which is that the loadDetaileldApp from the appcenter was not called